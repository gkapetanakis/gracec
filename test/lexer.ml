let () =
  let dirname = "programs/" in
  let filenames = [ "program1.grc" ] in
  let test filename =
    let chan = open_in (dirname ^ filename) in
    let lexbuf = Lexing.from_channel chan in
    let rec tokenize_and_print () =
      try
        (* call the lexer's 'token' rule (tokenize input) *)
        let token = Gracec_lib.Lexer.token lexbuf in
        (* print found tokens *)
        match token with
        | Gracec_lib.Tokens.EOF -> ()
        | _ ->
            print_endline (Gracec_lib.Lexer.string_of_token token);
            tokenize_and_print ()
      with Gracec_lib.Lexer.Lexing_error msg -> print_endline msg
    in
    tokenize_and_print ()
  in
  List.iter test filenames

(*
  while true do
    print_endline "Give some input, or use Ctrl+C to quit:";
    let input_string = read_line () in
    let lexbuf = Lexing.from_string input_string in
  done
*)
