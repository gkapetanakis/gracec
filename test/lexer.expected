fun
id (main)
lparen
rparen
colon
nothing
var
id (a)
colon
int
semicolon
var
id (b)
colon
char
semicolon
var
id (c)
colon
int
lbracket
int_lit (5)
rbracket
semicolon
var
id (d)
colon
char
lbracket
int_lit (10)
rbracket
semicolon
lbrace
rbrace
